dig.go

```diff
@@ -205,6 +207,36 @@ func New(opts ...Option) *Container {
 	return c
 }
 
+// A VisualizeOption modifies the default behavior of Visualize. It's included
+// for future functionality; currently, there are no concrete implementations.
+type VisualizeOption interface {
+	unimplemented()
+}
+
+// Visualize parses the graph in Container c into DOT format and writes it to
+// io.Writer w.
+func Visualize(c *Container, w io.Writer, opts ...VisualizeOption) error {
+	graphTmpl := template.Must(template.New("DotGraph").Parse(`digraph {
+	graph [compound=true];
```

Comment:
With the `template.Must` business, I suggest putting it in a package-level variable so it happens only once.

```go
var _graphTmpl = template.Must(...)

func Visualize(c *Container, ...) error {
  if err := graphTmpl.Execute(w, c.dg); err != nil {
    ...
  }
  return nil
}
```

Reply:
Done!